### PR TITLE
Agreements View and create | psp-5622 & psp-4717

### PR DESCRIPTION
**Pull request by FuriousLlama**
image
image

### CODE REVIEW

File: source/database/mssql/scripts/dbscripts/PSP_PIMS_S51_00/Alter Down/076_DML_PIMS_ROLE_CLAIM.sql
Comment (FuriousLlama) [inline source/database/mssql/scripts/dbscripts/PSP_PIMS_S51_00/Alter Down/076_DML_PIMS_ROLE_CLAIM.sql:5]: This PR updates the corresponding claims for agreements

**Comment by devinleighsmith** [comment]
Hmm, the only requirements I see for no agreements is on the uxpin which is this:
image

but that doesn't match with the implementation.

**Comment by devinleighsmith** [comment]
Confluence lists purchase prices as a required field but it is not required in the implementation. Also the uxpin doesn't have this as required which is confusing.

**Comment by devinleighsmith** [comment]
mismatch between confluence and uxpin:
Deposit due (days)
vs
deposit no later than days

also the implementation doesn't match confluence or uxpin, so probably at least need some doc updates so that we don't get bugs logged for this.

**Comment by devinleighsmith** [comment]
also deposit(confluence) vs deposit amount(uxpin)

**Comment by devinleighsmith** [comment]
nit: maybe a difference in vertical spacing between sections in uxpin vs. implementation? not sure if worth fixing, could ask Bobbi.

**Comment by devinleighsmith** [comment]
Not seeing confirmation of deletion as per this AC on psp-4717:
Ability to remove agreements.
Please confirm the action of the user before removing.

**Comment by devinleighsmith** [comment]
when changing the value of the agreement type the commencement date field is hidden but not cleared. Normally we try and clear hidden fields - this isn't called out in the jira but it probably should be.

File: source/backend/api/Models/Concepts/AcquisitionFile/AgreementModel.cs
Comment (devinleighsmith) [inline source/backend/api/Models/Concepts/AcquisitionFile/AgreementModel.cs:27]: right, so I guess this should be non-nullable as per confluence.

File: source/backend/dal/Repositories/AgreementRepository.cs
Comment (devinleighsmith) [inline source/backend/dal/Repositories/AgreementRepository.cs:34]: is this used anywhere? also, is this repository a good place for this?

File: source/backend/dal/Repositories/AgreementRepository.cs
Comment (devinleighsmith) [inline source/backend/dal/Repositories/AgreementRepository.cs:45]: do we have a pattern around using this? it seems odd to have this on just one of the repo methods.

File: source/frontend/src/features/mapSideBar/tabs/SectionStyles.tsx
Comment (devinleighsmith) [inline source/frontend/src/features/mapSideBar/tabs/SectionStyles.tsx:68]: thanks for centralizing these.

File: source/frontend/src/features/properties/map/acquisition/ViewSelector.tsx
Comment (devinleighsmith) [inline source/frontend/src/features/properties/map/acquisition/ViewSelector.tsx:62]: could we do null instead of -1?

File: source/frontend/src/features/properties/map/acquisition/add/AddAcquisitionContainer.test.tsx
Comment (devinleighsmith) [inline source/frontend/src/features/properties/map/acquisition/add/AddAcquisitionContainer.test.tsx:65]: thanks, this is preferred.

File: source/frontend/src/features/properties/map/acquisition/detail/checklist/AcquisitionChecklistView.tsx
Comment (devinleighsmith) [inline source/frontend/src/features/properties/map/acquisition/detail/checklist/AcquisitionChecklistView.tsx:79]: nit: in the future can we try and keep enhancement PRs like this separate? I don't want to whine too much because I like the improvements, but checklists don't really have much to do with agreements.

File: source/frontend/src/features/properties/map/agreement/detail/AgreementContainer.tsx
Comment (devinleighsmith) [inline source/frontend/src/features/properties/map/agreement/detail/AgreementContainer.tsx:16]: test coverage?

File: source/frontend/src/features/properties/map/agreement/detail/AgreementView.test.tsx
Comment (devinleighsmith) [inline source/frontend/src/features/properties/map/agreement/detail/AgreementView.test.tsx:13]: does the view need to mock the api?

File: source/frontend/src/features/properties/map/agreement/detail/AgreementView.tsx
Comment (devinleighsmith) [inline source/frontend/src/features/properties/map/agreement/detail/AgreementView.tsx:38]: think you are missing a key here.

File: source/frontend/src/features/properties/map/agreement/update/AgreementSubForm.tsx
Comment (devinleighsmith) [inline source/frontend/src/features/properties/map/agreement/update/AgreementSubForm.tsx:47]: hmm, perhaps this isn't working then? didn't seem to be resetting the field in my tests.

Comment (devinleighsmith) [reply source/frontend/src/features/properties/map/agreement/update/AgreementSubForm.tsx:47]: also seems like a good unit test candidate.

File: source/frontend/src/features/properties/map/agreement/update/UpdateAgreementsYupSchema.ts
Comment (devinleighsmith) [inline source/frontend/src/features/properties/map/agreement/update/UpdateAgreementsYupSchema.ts:26]: nit: could be something like "Invalid number"

File: source/frontend/src/models/api/Agreement.ts
Comment (devinleighsmith) [inline source/frontend/src/models/api/Agreement.ts:16]: would affect this as well

**Review by devinleighsmith — APPROVED** [review]

File: source/frontend/src/features/properties/map/acquisition/ViewSelector.tsx
Comment (FuriousLlama) [reply source/frontend/src/features/properties/map/acquisition/ViewSelector.tsx:62]: well it should never be either -1 or null. What we need is to refactor the viewSelector to avoid passing acq files with unknown ids

File: source/frontend/src/features/properties/map/acquisition/detail/checklist/AcquisitionChecklistView.tsx
Comment (FuriousLlama) [reply source/frontend/src/features/properties/map/acquisition/detail/checklist/AcquisitionChecklistView.tsx:79]: Yeah I agree. I was centralizing the styles and refactored some of the checklist code

File: source/backend/api/Models/Concepts/AcquisitionFile/AgreementModel.cs
Comment (FuriousLlama) [reply source/backend/api/Models/Concepts/AcquisitionFile/AgreementModel.cs:27]: Until I get a response from the BAs I will merge this and fix it if they requrie it.

File: source/frontend/src/features/properties/map/agreement/update/AgreementSubForm.tsx
Comment (FuriousLlama) [reply source/frontend/src/features/properties/map/agreement/update/AgreementSubForm.tsx:47]: odd, it works when I tried it

File: source/backend/dal/Repositories/AgreementRepository.cs
Comment (FuriousLlama) [reply source/backend/dal/Repositories/AgreementRepository.cs:45]: I've seen it too in some repositories but not all.

File: source/backend/dal/Repositories/AgreementRepository.cs
Comment (FuriousLlama) [reply source/backend/dal/Repositories/AgreementRepository.cs:34]: bad copypasta

**Pull request merged**
